### ISSUE TITLE
Reproducibility documentation should be updated

### ISSUE THREAD

**Issue by XiaoXiao-Woo** [pytorch/pytorch#104033]
### 📚 The doc issue

When loading the weights, the model and the optimizer are loaded. Besides, the sampled data in dataloader are guaranteed to be the same. It should be considered in https://pytorch.org/docs/stable/notes/randomness.html.

### Suggest a potential alternative/fix

import torch
import numpy
import random
print(torch.__version__)  # 1.13.0+cu116

def set_random_seed(seed, deterministic=True):
    """Set random seed.

    Args:
        seed (int): Seed to be used.
        deterministic (bool): Whether to set the deterministic option for
            CUDNN backend, i.e., set `torch.backends.cudnn.deterministic`
            to True and `torch.backends.cudnn.benchmark` to False.
            Default: False.
    """
    random.seed(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    torch.cuda.manual_seed(seed)
    torch.cuda.manual_seed_all(seed)
    if deterministic:
        torch.backends.cudnn.deterministic = True
        torch.backends.cudnn.benchmark = False

if __name__ == '__main__':

    n = 100
    set_random_seed(1)

    seed = 1#int(torch.empty((), dtype=torch.int64).random_().item())
    generator = torch.Generator()

    # Function 'torch.Generator()' is not limited by the global seed
    # because there is another seed offset inside it to ensure that each epoch is different
    generator.manual_seed(seed)

    def simple_generator():
        # When the epoch is increasing, torch.randperm also generates a seed offset inside,
        # which can be controlled by the generator.
        # Note that it cannot be set the same for each epoch

        # When loading the weights, the model and the optimizer are loaded.
        # Besides, the sampled data in dataloader are guaranteed to be the same.
        # See Line: 36-39
        yield torch.randperm(n, generator=generator).tolist()

    state = generator.get_state()

    for i in range(3):
        my_gen = simple_generator()
        # It will be make sure each epoch is the same, but don't do this,
        # only when restoring the model to ensure that the sampled data are the same.
        # It should be considered in https://pytorch.org/docs/stable/notes/randomness.html
        print(generator.set_state(state))
        print(generator.get_state().sum())
        # print(torch.get_rng_state().sum())
        print(next(my_gen))

    # It is related to the instance, not the global.
    print(torch.Generator().get_state().sum())
